src/agent/onefuzz-agent/src/tasks/coverage/recorder.rs

```diff
@@ -22,17 +20,71 @@ use crate::tasks::coverage::libfuzzer_coverage::Config;
 
 pub struct CoverageRecorder {
     config: Arc<Config>,
-    script_dir: OwnedDir,
+    script_path: PathBuf,
+    // keep _temp_dir such that Drop cleans up temporary files
+    _temp_dir: Option<TempDir>,
 }
 
 const SYMBOL_EXTRACT_ERROR: &str = "Target appears to be missing sancov instrumentation. This error can also happen if symbols for the target are not available.";
 
 impl CoverageRecorder {
-    pub fn new(config: Arc<Config>) -> Self {
-        let script_dir =
-            OwnedDir::new(env::var("ONEFUZZ_TOOLS").unwrap_or_else(|_| "script".to_string()));
+    pub async fn new(config: Arc<Config>) -> Result<Self> {
+        let (script_path, _temp_dir) = match env::var("ONEFUZZ_TOOLS") {
+            Ok(tools_dir) => {
+                let script_path = PathBuf::from(tools_dir);
+                if cfg!(target_os = "linux") {
```

Comment:
It might be easier to track the logic if these branch cases were broken out into `#[cfg(target_os = "<os>")]`-annotated helper functions.